File: monorepo_csharp/apps/Web/Imports.cs
// 1. Plain namespace using.
using System;
// 3. using static.
using static System.Math;
// 4. Namespace alias.
using SysColls = System.Collections.Generic;
// 5. Type alias.
using StringList = System.Collections.Generic.List<string>;
// 6. Cross-workspace-package usings.
using Mono.Shared;
using static Mono.Utils.Clamp;

namespace Mono.Web;

public static class Imports
{
    public static void Demo()
    {
        Console.WriteLine("hello");
        Console.WriteLine(PI + Sqrt(2) + Max(1, 2));
        var q = new SysColls.Queue<int>();
        var names = new StringList();
        var u = new User(1, "alice");
        Console.WriteLine($"{Util.FormatUser(u)} q={q.Count} n={names.Count} c={Between(42, 0, 10)}");
    }
}

File: monorepo_kotlin/apps/web/src/main/kotlin/mono/web/Imports.kt
package mono.web

// 1. Single-class import.
import kotlin.math.PI
// 2. Wildcard.
import kotlin.collections.*
// 3. Aliased import.
import kotlin.math.sqrt as kSqrt
// 4. Top-level function import.
import kotlin.math.max
// 5. Re-imported with alias.
import kotlin.math.PI as MATH_PI
// 6. Cross-workspace-package imports.
import mono.shared.User
import mono.shared.Util
import mono.utils.TAG
import mono.utils.clamp as kclamp

fun importsDemo() {
    val xs = arrayListOf(1, 2, 3)
    val v = PI + kSqrt(2.0) + max(1, 2) + MATH_PI
    val u = User(1, "imports")
    println("$xs sum=${xs.sum()} v=$v")
    println("shared=${Util.formatUser(u)} tag=$TAG clamp=${kclamp(42, 0, 10)}")
}
